User Manual update -- secondary indexes

DLotts opened the pull request:
## Description
Added a manual page to cover secondary indexes

### Tests
Hmmm, not sure how that would work.  

### Links
TODO:  [Jira](https://issues.apache.org/jira/browse/RYA-NUMBER)

### Checklist
- [ ] Text Review
- [ ] Squash Commits

#### People To Reivew
Puja
Authors of the indexes

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/702/

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/704/

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/714/<h2>Build result: FAILURE</span></h2>[...truncated 234.53 KB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 06:59 min[INFO] Finished at: 2018-03-22T21:02:02+00:00[INFO] Final Memory: 203M/3162M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.manual: Too many files with unapproved license: 1 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/rya.manual/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.manualchannel stoppedSetting status of 00568c3f01e65ee507c6dd95bf6d6f9c5919a9f7 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/714/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

DLotts commented:
Sorry folks, I goofed.  I fixed a tangle when merging and did a forced
push, changing one (some?) of the recent commits.  You may experience some
issues if you pulled/cloned from master early this week or last week.

For some reason after I merged a pull request, I had two copies of an old
commit and a third commit that had no changes in it.  I have seen similar
and fixed this before on my own repo.

I fixed it within five minutes of breaking it, so I thought it would only
affect people that pulled within that window.  But I might be wrong.  The
commit labeled:

 "POM version fix. Closes #281"

has changed it's ID.  Also it appears *after* the commit I just merged.
Hopefully that is all that changed.

This should be only a minor annoyance for anyone that pulled/cloned from
master early this week or last week.  Let me know if I can help.

david.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/715/<h2>Build result: FAILURE</span></h2>[...truncated 234.53 KB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 06:45 min[INFO] Finished at: 2018-03-26T20:36:10+00:00[INFO] Final Memory: 140M/2838M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.manual: Too many files with unapproved license: 1 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/rya.manual/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.manualchannel stoppedSetting status of d13f9b9474c87baf5127c7a5db4ff658c61e5090 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/715/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/788/<h2>Build result: FAILURE</span></h2>[...truncated 643.40 KB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 14:49 min[INFO] Finished at: 2019-03-12T20:04:06+00:00[INFO] Final Memory: 204M/2990M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.manual: Too many files with unapproved license: 1 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/rya.manual/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.manualchannel stoppedSetting status of 62b74daa5e3d7028ea8c466dee0c39edd5373537 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/788/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/789/<h2>Build result: FAILURE</span></h2>[...truncated 630.90 KB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 12:58 min[INFO] Finished at: 2019-03-12T20:18:21+00:00[INFO] Final Memory: 158M/2556M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.manual: Too many files with unapproved license: 1 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests@2/extras/rya.manual/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.manualchannel stoppedSetting status of a8fb79e0ffa9f486558eb8395d44605dff304cbd to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/789/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing
